data_acquisition/src/NHD_Chebyshev_Center_Pull.py

```diff
@@ -91,44 +98,82 @@ def maximum_no_of_tasks(MaxNActive, waitingPeriod):
 
+
 def calc_area(feature):
     return(feature.set('area_calc',feature.area().divide(1e6)))
 
-assets_done = ee.data.listAssets({'parent': 'projects/earthengine-legacy/assets/users/sntopp/NHD/DeepestPoint'})['assets']
-ids_done = [i['id'].split('/')[-1] for i in assets_done]
-assets_parent = ee.data.listAssets({'parent': 'projects/earthengine-legacy/assets/projects/sat-io/open-datasets/NHD'})['assets']
-assets_parent = [i for i in assets_parent if i['id'].split('/')[-1] not in ids_done]
-#assets_parent = [i for i in assets_parent if i['id'].split('/')[-1] not in ['NHD_MO','NHD_TX','NHD_AK']]
 
+## Remove geometries
+def remove_geo(image):
+  """ Funciton to remove the geometry from an ee.Image
+  
+  Args:
+      image: ee.Image of an ee.ImageCollection
+      
+  Returns:
+      ee.Image with the geometry removed
+  """
+  return image.setGeometry(None)
 
-for i in range(len(assets_parent)):
-    state_asset = assets_parent[i]['id']
-    assets_state = (ee.FeatureCollection(f"{state_asset}/NHDWaterbody")
-    .filter(ee.Filter.gte('areasqkm',0.001))
-    .filter(ee.Filter.lte('areasqkm',5000))  #Remove Great Lakes
-    .filter(ee.Filter.inList('ftype',[361,436,390])))
 
-    ## Added to trouble shoot NM where some of the NHD areas are don't match the actual polygon areas
-    assets_state=ee.FeatureCollection(assets_state.map(calc_area)).filter(ee.Filter.gte('area_calc',0.001))
+def get_lat_long(feature):
+  lat_lon = feature.geometry().coordinates()
+  lat = lat_lon.get(1)
+  lon = lat_lon.get(0)
+  return (feature.set({
+    'latitude': lat,
+    'longitude': lon}))
 
-    dp = assets_state.map(GetLakeCenters)
 
-    dataOut = ee.batch.Export.table.toAsset(collection=dp,description=state_asset.split('/')[-1],assetId=f"projects/earthengine-legacy/assets/users/sntopp/NHD/DeepestPoint/{state_asset.split('/')[-1]}")
+## Get NHD Asset Lists
+state_assets = ee.data.listAssets({'parent': 'projects/sat-io/open-datasets/NHD'})['assets']
 
-    ## Check how many existing tasks are running and take a break if it's >15
-    maximum_no_of_tasks(15, 240)
-    ## Send next task.
-    dataOut.start()
-    print(state_asset.split('/')[-1])
 
-#f"projects/earthengine-legacy/assets/users/sntopp/NHD/{state_asset.split('/')[-1]}/NHDDeepestPoint"
+## Get State asset
+states = ee.FeatureCollection('TIGER/2018/States')
+
 
+## for each state, run each grid cell and then export to file
+for i in range(len(state_assets)):
+    state_asset = state_assets[i]['id']
+    state_waterbody = (ee.FeatureCollection(f"{state_asset}/NHDWaterbody")
+      .filter(ee.Filter.gte('areasqkm',0.01))
+      .filter(ee.Filter.lte('areasqkm',5000))  #Remove Great Lakes
+      .filter(ee.Filter.inList('ftype',[361,436,390]))) #Only grab lakes, ponds, and reservoirs
+    
+    ## Added to trouble shoot NM where some of the NHD areas are don't match the actual polygon areas
+    state_waterbody=ee.FeatureCollection(state_waterbody.map(calc_area)).filter(ee.Filter.gte('area_calc',0.001))
```

Comment:
What causes big differences? Are they reservoirs whose maintenance makes them experience yo-yo depths?

Reply:
ha! This is left over from simon - it's what i was trying to get around by doing things in 1deg tiles later.